src/lib/yang/alarms.lua

```diff
@@ -387,6 +424,11 @@ function set_operator_state (key, args)
       state = args.state,
       text = args.text,
    })
+   if args.state == 'shelved' then
+      shelve_alarm(key, alarm)
+   elseif args.state == 'un-shelved' then
+      unshelve_alarm(key, alarm)
```

Comment:
Is there a third case here?  If not, we should assert that state is either shelved or unshelved, or otherwise use a shelved boolean property.